# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 2a040986845d
# Full Hash: 2a040986845da97f1eaf87c33136d9168817606a
# Author: Neil Deakin <neil@mozilla.com>
# Date: 2015-04-13 10:00:52
# Regressor Bug: 1071562
# File Overlap Count: 1
# Description:
#   Bug 1071562, e10s, support non-text types in clipboard (html, images, etc), r=smaug
# ==============================================================================

diff -r 8975ed86f399 -r 2a040986845d dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Fri Apr 10 09:43:38 2015 -0700
+++ b/dom/ipc/PContent.ipdl	Fri Apr 10 12:51:06 2015 -0400
@@ -767,12 +767,22 @@
     // nsIPermissionManager messages
     sync ReadPermissions() returns (Permission[] permissions);
 
-    SetClipboardText(nsString text, bool isPrivateData, int32_t whichClipboard);
-    sync GetClipboardText(int32_t whichClipboard)
-        returns (nsString text);
-    EmptyClipboard(int32_t whichClipboard);
-    sync ClipboardHasText(int32_t whichClipboard)
-        returns (bool hasText);
+    // Places the items within dataTransfer on the clipboard.
+    SetClipboard(IPCDataTransfer aDataTransfer,
+                 bool aIsPrivateData,
+                 int32_t aWhichClipboard);
+
+    // Given a list of supported types, returns the clipboard data for the
+    // first type that matches.
+    sync GetClipboard(nsCString[] aTypes, int32_t aWhichClipboard)
+        returns (IPCDataTransfer dataTransfer);
+
+    // Clears the clipboard.
+    EmptyClipboard(int32_t aWhichClipboard);
+
+    // Returns true if data of one of the specified types is on the clipboard.
+    sync ClipboardHasType(nsCString[] aTypes, int32_t aWhichClipboard)
+        returns (bool hasType);
 
     sync GetSystemColors(uint32_t colorsCount)
         returns (uint32_t[] colors);